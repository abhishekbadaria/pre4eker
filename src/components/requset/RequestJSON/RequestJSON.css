.json {
  margin-bottom: .5rem;
}

.inline {
  display: flex;
  margin: 0 -.3em .5rem -.3em;
}

.half {
  width: 50%;
  padding: 1rem .3rem;
}

.third {
  width: 33%;
  padding: 1rem .3rem;
}

.inputGroup label {
  color: #fff;
}