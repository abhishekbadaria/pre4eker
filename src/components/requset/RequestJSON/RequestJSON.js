import React, { useState } from "react";
import PropTypes from "prop-types";

import AceEditor from "react-ace";

import "brace/mode/json";
import "brace/theme/xcode";

import Button from "src/components/Button/Button";
import Input from "src/components/Input/Input";

import style from "./RequestJSON.css";

const RequestJSON = ({ data, onSubmit }) => {
  const [body, setBody] = useState(JSON.stringify(data.body, null, 1));
  const [endpoint, setEndpoint] = useState(data.endpoint);
  const [operation, setOperation] = useState(data.operation);
  const [key, setKey] = useState(data.key);

  return (
    <div>
      <div className={`${style.inline} ${style.inputGroup}`}>
        <div className={style.third}>
          <Input
            label="Эндпоин"
            name="endpoint"
            onChange={(e) => setEndpoint(e.target.value)}
            value={endpoint}
          />
        </div>
        <div className={style.third}>
          <Input
            label="Секретный ключ"
            name="key"
            value={key}
            onChange={(e) => setKey(e.target.value)}
            type="password"
          />
        </div>
        <div className={style.third}>
          <Input
            label="Операция"
            name="operation"
            value={operation}
            onChange={(e) => setOperation(e.target.value)}
          />
        </div>
      </div>
      <AceEditor
        name="RequestBody"
        mode="json"
        theme="xcode"
        value={body}
        onChange={(value) => setBody(value)}
        height="58vh"
        width="100%"
        wrapEnabled
        placeholder="Тело ответа"
        highlightActiveLine={false}
        setOptions={{
          showLineNumbers: false,
          tabSize: 2,
        }}
      />
      <Button
        action={() =>
          onSubmit({
            endpoint,
            key,
            body,
            operation,
          })
        }
        type="TEXT"
        size="sizeFull"
      >
        Показать
      </Button>
    </div>
  );
};

RequestJSON.defaultProps = {
  data: {},
};

RequestJSON.propTypes = {
  data: PropTypes.object,
  onSubmit: PropTypes.func.isRequired,
};

export default RequestJSON;
