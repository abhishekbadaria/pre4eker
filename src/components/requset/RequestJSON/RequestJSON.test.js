import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { act } from "react-dom/test-utils";

import RequestJSON from "./RequestJSON";

describe("Request JSON form", () => {
  it("should show set data", () => {
    const setJSON = jest.fn();
    const handleSubmit = jest.fn();
    const expectText = { a: "b" };
    let shownText;

    const { getByText } = render(
      <RequestJSON
        setRequestJSON={setJSON}
        requestJSON={expectText}
        handleSubmit={handleSubmit}
      />
    );

    act(() => {
      window.editor.current.editor.setValue(expectText);
      console.log(window.editor.current.editor.container.innerText);
    });

    // expect(shownText).toBe('{"a": "b"}');
  });
});
