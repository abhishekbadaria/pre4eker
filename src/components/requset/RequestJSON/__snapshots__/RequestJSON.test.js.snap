// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Response form should handle click on btn 1`] = `
<DocumentFragment>
  <div
    class="inline inputGroup"
  >
    <div
      class="third"
    >
      <div
        class="inputItem"
      >
        <label
          class="label"
          for="endpoint"
        >
          Эндпоин
        </label>
        <input
          class="input"
          name="endpoint"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="third"
    >
      <div
        class="inputItem"
      >
        <label
          class="label"
          for="key"
        >
          Секретный ключ
        </label>
        <input
          class="input"
          name="key"
          type="password"
          value=""
        />
      </div>
    </div>
    <div
      class="third"
    >
      <div
        class="inputItem"
      >
        <label
          class="label"
          for="operation"
        >
          Операция
        </label>
        <input
          class="input"
          name="operation"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
  <div
    class="editorWrapper"
  >
    <div
      class="loading"
    >
      <div
        class="loader loader-undefined loader-active"
      >
        <div
          class="loader-inner ball-scale-ripple-multiple"
        >
          <div />
          <div />
          <div />
        </div>
      </div>
    </div>
  </div>
  <button
    class="sizeFull button normal false basic"
    type="button"
  >
    Рассчитать скидки
  </button>
</DocumentFragment>
`;
